### DISCUSSION TITLE
Using getStaticPaths with multiple dynamic routes

### DISCUSSION THREAD

**Discussion by yassinebridi** [vercel/next.js#13508] [Help]
I have a path like this, which is a pagination for the category pages:
```
pages/[category]/page/[id]
```
and i already created pages for `pages/[category]/index`, inside it's getStaticPaths like this:
```
export const getStaticPaths: GetStaticPaths = async () => {
  const posts = getAllPosts(["mainTag"]);

  return {
    paths: posts.map((post) => {
      return {
        params: {
          category: post.mainTag,
        },
      };
    }),
    fallback: false,
  };
};
```

so for the first route, i need [category] and [id], for the params. For the [id], it's simple:

```
export const getStaticPaths: GetStaticPaths = async () => {
  const posts = getAllPosts(["mainTag"]);
  const allPosts = posts.slice(22)
  const rowsPerPage = POSTPERPAGE
  const totalResult = allPosts.length
  const pageCount = Math.ceil(totalResult / rowsPerPage)

  return {
    paths: Array.from({ length: pageCount }).map((_, i) => {
      return {
        params: {
          id: (i + 1).toString(),
        },
      };
    }),
    fallback: false,
  };
};
```
this will create all the necessary pagination pages for category, **but i can't get the [category] dynamically here.**
I wonder if there is a way to share the category directly to the [id] page.

**Comment by net-caster** [comment]
@yassinebridi I don't know if you've figured this out yet but as per the [documentation](https://nextjs.org/docs/basic-features/data-fetching#the-paths-key-required) you need to define the id **and** the category in the paths:

```
return {
    paths: Array.from({ length: pageCount }).map((_, i) => {
      return {
        params: {
          id: (i + 1).toString(),
          category: categoryName
        },
      };
    }),
    fallback: false,
  };
};
```

I tried your approach (static/dynamic/static/dynamic) and it works.
